sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/ShuffledBatchRDD.scala

```diff
@@ -155,13 +155,37 @@ class ShuffledBatchRDD(
   }
 
   override def compute(split: Partition, context: TaskContext): Iterator[ColumnarBatch] = {
-    val shuffledRowPartition = split.asInstanceOf[ShuffledBatchRDDPartition]
+    val shuffledBatchPartition = split.asInstanceOf[ShuffledBatchRDDPartition]
+    // update partition metrics
+    metrics(GpuMetricNames.NUM_PARTITIONS).add(1)
+    val partitionSize = shuffledBatchPartition.spec match {
```

Comment:
Nit: We're already matching on the partition spec below, and if a new spec is added we need to keep these two in sync.  Would it make more sense to compute the metric (or at least the part of the computation specific to the partition spec) in the existing match rather than adding a separate match?

Reply:
I actually started out doing it that way and then split it out. I wasn't sure which was more readable. I can move it back.